### PR TITLE
Add assemblyscript configuration in subgraph; fixed bug in handlePoolDetailsChanged; various requested changes

### PR DESCRIPTION
**Pull request by krisbitney**
incorporates changes requested in https://github.com/GoodDollar/GoodCollective/pull/60#pullrequestreview-1793073058

### CODE REVIEW

File: packages/subgraph/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+  "extends": "assemblyscript/std/assembly.json",
+  "include": [
+    "./src/**/*.ts",
+  ],
+}
```
Comment (sirpy) [inline packages/subgraph/tsconfig.json:6]: why is that needed?

Comment (krisbitney) [reply packages/subgraph/tsconfig.json:6]: Your IDE won't confuse TypeScript types and AssemblyScript types. For example,`i32` is a valid AssemblyScript primitive but `number` is not. Please review the Telegram conversation for more context.

**Comment by krisbitney** [comment]
replaced by https://github.com/GoodDollar/GoodCollective/pull/64
